### PR TITLE
Integrate following api

### PR DESCRIPTION
**Pull request by sharok**

### CODE REVIEW

File: client/controllers/user.js [hunk 1]
```diff
@@ -14,19 +16,25 @@ export default class UserController extends Controller {
...
     posts.order = '-created';
-
     this.xhrs.user = user.fetch();
     this.xhrs.posts = posts.fetch();
+    this.xhrs.followers = followers.fetchCount();
+    this.xhrs.followings = followings.fetchCount();
+
 
-    let dfd = $.when(this.xhrs.user, this.xhrs.posts);
+    let dfd = $.when(this.xhrs.followers, this.xhrs.user, this.xhrs.posts);
```
Comment (atomiomi) [inline client/controllers/user.js]: `this.xhrs.followings` ?

Comment (sharok) [inline client/controllers/user.js]: Forgot :)

File: client/models/followings.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import Collection from '../base/collection';
+
+
+export default class Followings extends Collection {
```
Comment (atomiomi) [inline client/models/followings.js:4]: Let's make same with `FollowingUser`, call it `FollowingUsers` (don't forget file name)

File: client/modules/user.js [hunk 3]
```diff
@@ -85,6 +85,7 @@ class User extends Model {
   authorized () {
     return !!this.id;
   }
+
```
Comment (atomiomi) [inline client/modules/user.js]: Don't commit empty lines

File: client/views/components/follow_toggle.js [hunk 4]
```diff
@@ -1,25 +1,49 @@
...
-      isFollow: false,
-    };
+
+  constructor () {
+    super();
+    this.handleToggle = _.debounce(this.handleToggle.bind(this), TOGGLE_LIMIT);
+  }
+
+  componentDidMount () {
+    var isFollowed = this.props.user.isFollowed;
```
Comment (atomiomi) [inline client/views/components/follow_toggle.js]: let

File: client/views/components/follow_toggle.js [hunk 5]
```diff
@@ -1,25 +1,49 @@
...
+    this.setState({isFollowed});
   }
 
   handleToggle () {
-    // TODO: API integration
-    let toggle = !this.state.isFollow;
-    this.setState({isFollow: toggle});
+    let followingUser = new FollowingUser({_id: this.props.user.username});
+
+    followingUser.followeeUsername = this.props.user.username;
```
Comment (atomiomi) [inline client/views/components/follow_toggle.js]: user follows themselves?

File: client/modules/user.js (hunk 3 above)
Comment (sharok) [inline client/modules/user.js]: I think jscs have to catch this error. Fix this please.

File: client/views/components/follow_toggle.js (hunk 4 above)
Comment (sharok) [inline client/views/components/follow_toggle.js]: And this too.

Comment (atomiomi) [inline client/views/components/follow_toggle.js]: I tried to config it, but there is no options for it, but I think it's possible to add rule for empty lines

File: client/views/search.js [hunk 6]
```diff
@@ -28,6 +29,7 @@ export default class User extends Component {
           <ul className="m-user-list">
             {results.length
               ? results.map((user, index) => {
+                let isCurrentUser = user.username === currentUser.get('username');
                 return <li key={index} className="m-ul-item">
```
Comment (atomiomi) [inline client/views/search.js:34]: add one space line

File: client/controllers/user.js [hunk 7]
```diff
@@ -14,19 +16,25 @@ export default class UserController extends Controller {
     let { username } = ctx.params;
     let user = new UserModel();
     let posts = new PostsCollection();
+    let followers = new FollowersCollection();
+    let followings = new FollowingsCollection();
 
     user.username = username;
     posts.username = username;
     posts.order = '-created';
-
```
Comment (atomiomi) [inline client/controllers/user.js:21]: Don't remove empty lines

File: client/views/components/follow_toggle.js [hunk 8]
```diff
@@ -1,25 +1,49 @@
 import React from 'react';
 import Component from '../../base/component';
+import FollowingUser from '../../models/following_user';
 import lang from '../../lang';
+import _ from 'lodash';
```
Comment (atomiomi) [inline client/views/components/follow_toggle.js]: vendor modules should be always at the top

File: client/views/components/follow_toggle.js (hunk 5 above)
Comment (sharok) [inline client/views/components/follow_toggle.js]: `followingUser.followeeUsername = this.props.user.username;`
This is artifact. I will delete it. I don't remember why I added it

File: client/models/followings.js (hunk 2 above)
Comment (sharok) [inline client/models/followings.js:4]: I think it's different. We have two collections: Followings and Followers. And one model FollowingUser. I think FollowingUser is not the same as Followings. The FollowingUser for me is just model for some action.

Comment (atomiomi) [inline client/models/followings.js:4]: So how do you look on renaming `FollowingUser` to `Followee`?

Comment (sharok) [inline client/models/followings.js:4]: Additional explanation. We can get collection of users who follows me, for this purpose we have Followings collection. But, if I want to get only one , I think we have to have another model, with name of Following.

Comment (sharok) [inline client/models/followings.js:4]: > So how do you look on renaming FollowingUser to Followee?

Not so good too. ;)
https://en.wiktionary.org/wiki/followee

Comment (atomiomi) [inline client/models/followings.js:4]: Ok, I like `Following` name

Comment (atomiomi) [inline client/models/followings.js:4]: Why?

> One who is followed (has his/her posts monitored by another user).

in this case user who is followed by current user

Comment (sharok) [inline client/models/followings.js:4]: Agree

File: client/views/components/follow_toggle.js [hunk 9]
```diff
@@ -1,25 +1,47 @@
...
+  componentDidMount () {
+    let isFollowed = this.props.user.isFollowed;
+    this.setState({isFollowed});
   }
 
   handleToggle () {
-    // TODO: API integration
-    let toggle = !this.state.isFollow;
-    this.setState({isFollow: toggle});
+    let followeeModel = new FolloweeModel({_id: this.props.user.username});
```
Comment (atomiomi) [inline client/views/components/follow_toggle.js]: just `followee` beacuse this is instance of model

File: client/base/collection.js [hunk 10]
```diff
@@ -23,7 +24,16 @@ export default class BaseCollection extends Collection {
...
+  fetchCount () {
+    let dfd = this.$(`${this.url()}/count`);
+    dfd.done((resp) => {
+      this.count = resp;
+    });
+
+    return dfd;
+  }
+}
+BaseCollection.prototype.$ = $.ajax;
```
Comment (atomiomi) [inline client/base/collection.js:38]: Space Line. There always should be space line after end before class definition

Comment (sharok) [inline client/base/collection.js:38]: Add all this shit to the codestyle guide

Comment (atomiomi) [inline client/base/collection.js:38]: Ok)

**Pull request merged**